### PR TITLE
feat: Add allowed origins config

### LINKED ISSUE #1355: Add configuration for allowed origins
Currently, if users start developing web apps that access an instance of DefraDB directly, they will be confronted with CORS errors from the browser because the app and DefraDB will operate on different ports locally and possibly different domains publicly. 

We should allow configuration of the HTTP API with a list of allowed origins.

*Note*: `WithAllowedOrigins` already exists on the API but we need to add the configuration through the CLI and config files.

### PR DESCRIPTION
**Pull request by fredcarle**
## Relevant issue(s)

Resolves #1355 

## Description

This PR adds the option to set allowed origins for the HTTP API using the config file or a CLI flag.

## Tasks

- [x] I made sure the code is well commented, particularly hard-to-understand areas.
- [x] I made sure the repository-held documentation is changed accordingly.
- [x] I made sure the pull request title adheres to the conventional commit style (the subset used in the project can be found in [tools/configs/chglog/config.yml](tools/configs/chglog/config.yml)).
- [x] I made sure to discuss its limitations such as threats to validity, vulnerability to mistake and misuse, robustness to invalidation of assumptions, resource requirements, ...

## How has this been tested?

Unit tests and curl with allowedOrigins set to http://test.com
```shell
# this will result in an unsuccessful Preflight request
curl -i http://localhost:9181/api/v1/ping -H "Origin: http://no.com" -H "Access-Control-Request-Method: GET" -X OPTIONS

# this will result in an successful Preflight request
curl -i http://localhost:9181/api/v1/ping -H "Origin: http://test.com" -H "Access-Control-Request-Method: GET" -X OPTIONS
```

Specify the platform(s) on which this was tested:
- MacOS

### CODE REVIEW

**Comment by AndrewSisley** [comment]
> Note: The default has been set to * which will allow all origins to send requests to the API. I feel like that's a good default for development but not for prod for obvious reasons. Do we want to be restrictive by default or open by default?

My vote would be for the opposite, and have it locked down by default. Forgetting (or not knowing about this) could be an expensive mistake. I also think that locked down by default is the norm (I remember having to mess with stuff like this a few times to enable CORS, but never to block it).

**Review by AndrewSisley — APPROVED** [review]
LGTM.  Approving assuming that the default value conversation is resolved before merged :)

**Comment by fredcarle** [comment]
> My vote would be for the opposite, and have it locked down by default. Forgetting (or not knowing about this) could be an expensive mistake. I also think that locked down by default is the norm (I remember having to mess with stuff like this a few times to enable CORS, but never to block it).

I'm happy with that 👍

File: README.md
Comment (shahzadlone) [inline README.md:382]: suggestion:

How about just `origins`. Mostly because we don't have a `blockedorigins` or any other option like that. Also `allowedorigins` is less readable, perhaps `allowedOrigins` if you want to stick with the "allowed" word.

```suggestion
defradb start --allowedorigins=https://yourdomain.com
```

Comment (fredcarle) [reply README.md:382]: The thing with `--origins` is that it's a little vague and the http header is called `Allowed-Origins`. I feel like it makes more sense to have the full representation here as well. `allowedOrigins` could be nicer.

Comment (AndrewSisley) [reply README.md:382]: nitpick: I have a very minor preference for `allowed-origins`, but only if it we dont already camel case any other existing multi-word params/flags

Comment (shahzadlone) [reply README.md:382]: I do like `allowed-origins`, would prefer that over `allowedOrigins`

**Review by shahzadlone — APPROVED** [review]
LGTM

File: README.md
Comment (orpheuslummis) [reply README.md:382]: personally i'd like `--api.corsdomain` or `http.corsdomain` depending on which top-level key we want to represent the http system

Comment (fredcarle) [reply README.md:382]: I'd rather we stay closer to what it represents. `allowed-origins` is quite good.

Comment (orpheuslummis) [reply README.md:382]: so the fully qualified key from the config's perspective would be `api.allowed-origins`, and as an env. variable it would be `DEFRA_API_ALLOWED_ORIGINS` perhaps. i'm not sure yet about if it's necessary to quote such a key with a `-` in Yaml  - it might have to be

```yaml
[api]
"allowed-origins": ...
```

Comment (fredcarle) [reply README.md:382]: I don't think it is according to what I see here https://www.cloudbees.com/blog/yaml-tutorial-everything-you-need-get-started

**Pull request merged**
